src/cc/usdt/usdt.cc

```diff
@@ -193,40 +212,87 @@ bool Probe::usdt_getarg(std::ostream &stream) {
   return true;
 }
 
+uint64_t Probe::usdt_getargfmt() {
+  assert(!locations_.empty());
+  if (argfmt_ == (uint64_t)-1) {
+    argfmt_ = 0;
+    const size_t arg_count = locations_[0].arguments_.size();
+
+    if (arg_count > 0) {
+      assert(arg_count <= 16);
```

Comment:
why 16? derived from some headers?

Reply:
No, it's my encoding of argument type that takes 4 bits so in a 64 bit variable, I can only encode up to 16 arguments. I suppose that should be enough. It's not the prettiest but is faster when comparing things. I can remove this and just use bigger structures too.